libs/@local/status/crate/src/status_code.rs

```diff
@@ -0,0 +1,169 @@
+// Attribution: *Heavily* inspired by the Google Cloud API Error Model
+//  https://cloud.google.com/apis/design/errors
+
+use std::fmt::{Display, Formatter, Result};
+
+use serde::{Deserialize, Serialize};
+
+mod http_compat;
+
+/// The canonical status codes for software within the HASH ecosystem.
+///
+/// Sometimes multiple status codes may apply. Services should return the most specific status code
+/// that applies. For example, prefer [`StatusCode::OutOfRange`] over
+/// [`StatusCode::FailedPrecondition`] if both codes apply. Similarly prefer
+/// [`StatusCode::NotFound`] or [`StatusCode::AlreadyExists`] over
+/// [`StatusCode::FailedPrecondition`].
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum StatusCode {
+    /// Not an error; returned on success.
+    ///
+    /// HTTP Mapping: 200 OK
+    Ok,
+
+    /// The operation was cancelled, typically by the caller.
+    ///
+    /// HTTP Mapping: 499 Client Closed Request
+    Cancelled,
+
+    /// Unknown error. For example, this error may be returned when a [`StatusCode::Status`] value
+    /// received from another address space belongs to an error space that is not known in this
+    /// address space. Also errors raised by APIs that do not return enough error information
+    /// may be converted to this error.
+    ///
+    /// HTTP Mapping: 500 Internal Server Error
+    ///
+    /// [`StatusCode::Status`]: crate::Status
```

Comment:
The intra-doc link is wrong
```suggestion
    /// [`Status`]: crate::Status
```

Reply:
[Fix broken doc links](https://github.com/hashintel/hash/pull/1787/commits/e63947f59f1a6ab1482e617cdece10dcaf4bce18)